### PR TITLE
InfluxDB: interpolate all variables in queries for dash->explore (for influxql)

### PR DESCRIPTION
**Pull request by joey-grafana**
**What this PR does / why we need it**:
This PR ensures that all variables are interpolated in queries when going from the dashboard to explore (for influxql). 

Previously we only interpolated query, measurement, policy and tags.

**How to test**
1. Create a dashboard panel for influxql
2. Edit the panel, adding variables to all parts of the query
3. Save, go back to the dash
4. Hover over the panel and click x to render the query in explore
5. All variables should be interpolated and the query should succeed

### CODE REVIEW

File: public/app/plugins/datasource/influxdb/datasource.ts [hunk 1]
```diff
@@ -340,17 +340,51 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
...
+        if (this.isFlux) {
+          return {
+            ...query,
+            query: this.templateSrv.replace(query.query ?? '', scopedVars, 'regex'),
+          };
+        }
+
         const expandedQuery = {
           ...query,
           datasource: this.getRef(),
```
Comment (toddtreece) [inline public/app/plugins/datasource/influxdb/datasource.ts:352]: i am guessing that the other pieces aren't important for `this.isFlux` queries, but is this change to `datasource` needed above in the flux query?

**Review by toddtreece — APPROVED** [review]
one open question, but this looks good to me if the `datasource` change isn't needed

File: public/app/plugins/datasource/influxdb/datasource.ts (hunk 1 above)
Comment (gabor) [reply public/app/plugins/datasource/influxdb/datasource.ts:352]: hmm.. i have no idea, but i'd say let's put that `datasource: this.getRef()` in the flux-branch too, to be safe.

**Comment by gabor** [comment]
@joey-grafana could you please try this scenario?

create a dashboard variable, name=`zone`, type=custom, value=`UTC, Europe/Berlin`

now, in the influxql query editor, set `${zone}` as the timezone.
switch to `Europe/Berlin`. this works.
now do the explore-this-query thing. now the timezone becomes `Europe\/Berlin`, which does not work.

i'm not sure where the problem is, maybe the problem is out of scope of this issue, not sure.

File: public/app/plugins/datasource/influxdb/datasource.ts (hunk 1 above)
Comment (joey-grafana) [reply public/app/plugins/datasource/influxdb/datasource.ts:352]: It was there from before the migration. It appears that it's not even necessary as `...query` already has the datasource ref. Adding it in for Flux just to be safe.

**Comment by joey-grafana** [comment]
> @joey-grafana could you please try this scenario?
> 
> create a dashboard variable, name=`zone`, type=custom, value=`UTC, Europe/Berlin`
> 
> now, in the influxql query editor, set `${zone}` as the timezone. switch to `Europe/Berlin`. this works. now do the explore-this-query thing. now the timezone becomes `Europe\/Berlin`, which does not work.
> 
> i'm not sure where the problem is, maybe the problem is out of scope of this issue, not sure.

Good catch. Removed the regex format, now it's using glob for the formatter and is working.
glob format

regex format

**Review by gabor — APPROVED** [review]
LGTM!

**Pull request merged**